### PR TITLE
Positivitize presumed positive quantity.

### PR DESCRIPTION
**Pull request by MTCam**

### CODE REVIEW

File: test/test_euler.py [hunk 1]
```diff
@@ -371,7 +371,7 @@ def inf_norm(data):
         # (Explanation courtesy of Mike Campbell,
         # https://github.com/illinois-ceesd/mirgecom/pull/44#discussion_r463304292)
 
-        momerr = inf_norm(iff_split.momentum) - p0
+        momerr = abs(inf_norm(iff_split.momentum) - p0)
```
Comment (thomasgibson) [inline test/test_euler.py:374]: `p0` is just a `DOFArray` right? How about we just write `inf_norm(iff_split.momentum - p0)`?

Comment (MTCam) [reply test/test_euler.py:374]: Yeah I thought about this.  The thing is that the momentum component here is a ndim x ndim array with only the diagonal being p0, so the off-diag term'll fail the test.

Comment (inducer) [inline test/test_euler.py:374]:
```suggestion
        momerr = inf_norm(abs(iff_split.momentum) - p0)
```

File: test/test_euler.py [hunk 2]
```diff
@@ -397,7 +397,7 @@ def inf_norm(data):
         assert inf_norm(bf_split.energy) < tolerance
         assert inf_norm(bf_split.species_mass) < tolerance
 
-        momerr = inf_norm(bf_split.momentum) - p0
+        momerr = abs(inf_norm(bf_split.momentum) - p0)
```
Comment (inducer) [inline test/test_euler.py:400]:
```suggestion
        momerr = inf_norm(abs(bf_split.momentum) - p0)
```

**Comment by MTCam** [comment]
I think it is what it should have been all along now.  Generate and compare against the exact expected answer. (e750535)

File: test/test_euler.py [hunk 3]
```diff
@@ -397,7 +401,12 @@ def inf_norm(data):
         assert inf_norm(bf_split.energy) < tolerance
         assert inf_norm(bf_split.species_mass) < tolerance
 
-        momerr = inf_norm(bf_split.momentum) - p0
+        # generate the exact answer: just p0*nhat for the given boundary
+        nhat = thaw(actx, discr.normal(BTAG_ALL))
+        expected_mom_flux = p0*nhat
+        mom_flux_exact = discr.project(BTAG_ALL, "all_faces", expected_mom_flux)
```
Comment (thomasgibson) [inline test/test_euler.py:407]:
```suggestion
        nhat = thaw(actx, discr.normal(BTAG_ALL))
        mom_flux_exact = discr.project(BTAG_ALL, "all_faces", p0*nhat)
```

File: test/test_euler.py [hunk 4]
```diff
@@ -371,7 +371,12 @@ def inf_norm(data):
         # (Explanation courtesy of Mike Campbell,
         # https://github.com/illinois-ceesd/mirgecom/pull/44#discussion_r463304292)
 
-        momerr = inf_norm(iff_split.momentum) - p0
+        # generate the exact answer: just p0*nhat for the given boundary
+        nhat = thaw(actx, discr.normal("int_faces"))
+        expected_mom_flux = p0*nhat
+        mom_flux_exact = discr.project("int_faces", "all_faces", expected_mom_flux)
```
Comment (thomasgibson) [inline test/test_euler.py:377]:
```suggestion
        nhat = thaw(actx, discr.normal("int_faces"))
        mom_flux_exact = discr.project("int_faces", "all_faces", p0*nhat)
```

**Review by thomasgibson — APPROVED** [review]
Super duper minor suggestions, but LGTM!

**Review by inducer — APPROVED** [review]
Thanks!

**Pull request merged**